### ISSUE TITLE
Add a way to anchor pattern fill to a column/bar point

### ISSUE THREAD

**Issue by AndrzejBuleczka** [highcharts/highcharts#23567]
# Description of the feature
Add anchorToPoint options to pattern fill: the pattern should be rendered identically within each column, as if each column has its own instance of the pattern anchored at its top-left corner (or center).

# Library related to the feature
Highcharts, pattern-fill module

# Proof of Concept/Live example for the feature
Here is the the current way the pattern fill works: https://jsfiddle.net/BlackLabel/kxtwbydr/

----

**You can vote for this feature by adding a thumbs-up reaction to this post.**

**Comment by jacob1474** [comment]
### Problematic behavior
The position of the pattern is not anchored to the individual column it fills. Instead, it appears to be rendered on a single, continuous canvas behind the columns. This causes the pattern to be cropped differently for each column depending on its position on the x-axis, leading to a visually inconsistent and unpredictable appearance. Some patterns can be not visible at all, if columns are too narrow, e.g. in this live [demo](https://jsfiddle.net/8t23oucf/) pattern for column `Test 2` is very hardly visible if it is visible at all. This violates WCAG 2.1 accessibility standards (Rule 1.3.3), as the patterns are intended to provide a reliable alternative to color for distinguishing data.

### Expected behavior
The pattern should be rendered identically within each column, as if each column has its own instance of the pattern anchored at its top-left corner (or center).

### Actual behavior
 I assume that the pattern is rendered on a global grid, causing its position relative to each column's boundaries to be inconsistent and not visible in some cases.

### Affected versions
Highcharts v12.4

### Affected browsers
All tested browsers (e.g., Chrome, Firefox, Edge, Brave)

**Comment by oysteinmoseng** [comment]
Good feedback! Image patterns already get positioned to the bounding box of the point, if set on a point. We should at minimum do the same with SVG patterns. Meaning, if you set a pattern on an individual point (not a series), it would be adjusted to the specific point.

Once we have this working we can consider if there should be some magic for certain series types where it makes sense to do this automatically even if you set the pattern on the whole series, but I think that should be step #2.

Note that as a workaround, sometimes you can get away with adjusting the pattern offset (or scaling) to achieve a result that works across points: https://jsfiddle.net/highcharts/ozsmfxbr/
